.github/workflows/include-check-cache.yml

```diff
@@ -104,349 +91,341 @@ jobs:
+    steps: 
     - name: Check JNI MacOS lib cache 
       id: jni-macos-lib-cache
-      uses: cmelchior/cache@main
+      uses: actions/cache@v4
       with:
         path: ./packages/cinterop/build/realmMacOsBuild
-        key: jni-macos-lib-${{ steps.packages-cache-key.outputs.sha }}
-
+        key: jni-macos-lib-${{ needs.checksum.outputs.packages-sha }}
     - name: Save JNI MacOS lib package
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       if: always() && !cancelled() && steps.jni-macos-lib-cache.outputs.cache-hit == 'true'
       with:
-        name: jni-macos-lib-${{ steps.find-library-version.outputs.label }}
+        name: jni-macos-lib-${{ needs.checksum.outputs.version-label }}
         path: ./packages/cinterop/build/realmMacOsBuild/**/*
         retention-days: 1
 
-    - name: Delete downloaded JVM cache files
-      id: delete-cache-macos-lib
-      uses: JesseTG/rm@v1.0.3
-      if: always() && !cancelled() && steps.jni-macos-lib-cache.outputs.cache-hit == 'true'
-      with:
-        path: ./packages/cinterop/build/realmMacOsBuild
+  #
+  # JNI Linux Lib
+  #    
+  jni-linux-lib:
+    runs-on: ubuntu-latest
+    needs: checksum
+    name: JNI Linux Lib
+    outputs:  
+      jni-linux-lib-cache-hit: ${{ steps.jni-linux-lib-cache.outputs.cache-hit }}
 
-    #
-    # JNI Linux Lib
-    #    
+    steps:
     - name: Check JNI Linux lib cache 
       id: jni-linux-lib-cache
-      uses: cmelchior/cache@main
+      uses: actions/cache@v4
       with:
         path: ./packages/cinterop/build/realmLinuxBuild
-        key: jni-linux-lib-${{ steps.packages-cache-key.outputs.sha }}
+        key: jni-linux-lib-${{ needs.checksum.outputs.packages-sha }}
 
     - name: Save JNI Linux lib package
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       if: always() && !cancelled() && steps.jni-linux-lib-cache.outputs.cache-hit == 'true'
       with:
-        name: jni-linux-lib-${{ steps.find-library-version.outputs.label }}
+        name: jni-linux-lib-${{ needs.checksum.outputs.version-label }}
         path: ./packages/cinterop/build/realmLinuxBuild/**/*
         retention-days: 1
 
-    - name: Delete downloaded JVM cache files
-      id: delete-cache-linux-lib
-      uses: JesseTG/rm@v1.0.3
-      if: always() && !cancelled() && steps.jni-linux-lib-cache.outputs.cache-hit == 'true'
+  #
+  # JNI Windows Lib
+  #    
+  jni-windows-lib:
+    runs-on: ubuntu-latest
+    needs: checksum
+    name: JNI Windows Lib
+    outputs:  
+      jni-windows-lib-cache-hit: ${{ steps.jni-windows-lib-cache.outputs.cache-hit }}
+    steps:
+    - name: Check JNI Windows lib cache 
+      id: jni-windows-lib-cache
+      uses: actions/cache@v4
       with:
-        path: ./packages/cinterop/build/realmLinuxBuild
+        path: ./packages/cinterop/build/realmWindowsBuild
+        key: jni-windows-lib-${{ needs.checksum.outputs.packages-sha }}
+        enableCrossOsArchive: true
```

Comment:
I guess this is unnecessary for Linux (`enableCrossOsArchive`) since we build and retrieve on Ubuntu

Reply:
I think so.